runtime/dart_isolate.cc

```diff
@@ -280,8 +284,130 @@ std::weak_ptr<DartIsolate> DartIsolate::CreateRootIsolate(
   return (*root_isolate_data)->GetWeakIsolatePtr();
 }
 
+bool DartIsolate::CreatePlatformIsolate(Dart_Handle entry_point,
+                                        Dart_Port isolate_ready_port_id,
+                                        const std::string& debug_name) {
+  PlatformConfiguration* platform_config = platform_configuration();
+  FML_DCHECK(platform_config != nullptr);
+  PlatformIsolateManager* platform_isolate_manager =
+      platform_config->client()->GetPlatformIsolateManager();
+  if (platform_isolate_manager->IsShutdown()) {
+    // Not allowed to spawn new platform isolates.
+    return false;
+  }
+
+  Dart_EnterScope();
+  Dart_PersistentHandle entry_point_handle =
+      Dart_NewPersistentHandle(entry_point);
+  Dart_ExitScope();
+
+  Dart_Isolate parent_isolate = isolate();
+  Dart_ExitIsolate();  // Exit parent_isolate.
+
+  // PlatformConfigurationClient* client = platform_configuration()->client();
+  // RuntimeController* runtime = (RuntimeController*)(client);
+  // std::cout << "RuntimeController: " << (void*)runtime << std::endl;
+
+  // TODO: This doesn't work for platform isolates spawned from other child
```

Comment:
Would it make things simpler if spawning of platform isolates was only allowed from within the root isolate?

Reply:
Yeah, this would be a lot simpler. I suppose this is a reasonable restriction, at least for now.